pkg/registries/types/metrics.go

```diff
@@ -0,0 +1,101 @@
+package types
+
+import (
+	"context"
+	"errors"
+	"math"
+	"net/http"
+	"os"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/metrics"
+)
+
+// MetricsHandler wraps the registry Prometheus metrics.
+type MetricsHandler struct {
+	requestCounter    *prometheus.CounterVec
+	timeoutCounter    *prometheus.CounterVec
+	durationHistogram *prometheus.HistogramVec
+}
+
+// roundedDurationBuckets produces exponential buckets rounded to the next millisecond.
+// The smallest bucket contains all durations smaller than 0.1 seconds. The largest
+// bucket is set to the registry client timeout.
+func roundedDurationBuckets() []float64 {
+	buckets := prometheus.ExponentialBucketsRange(0.1, env.RegistryClientTimeout.DurationSetting().Seconds(), 16)
```

Comment:
Consider using a static value for upper bound to keep it comparable between installations
```suggestion
	buckets := prometheus.ExponentialBucketsRange(0.1, 20, 16)
```